pkg/scheduler/framework/runtime/framework_test.go

```diff
@@ -1243,40 +1244,105 @@ func TestPreFilterPlugins(t *testing.T) {
+				{
+					name: "success",
+				},
+				{
+					name: "error",
+					inj:  injectedResult{PreFilterStatus: int(framework.Error)},
+				},
+			},
+			wantPreFilterResult: nil,
+			wantSkippedPlugins:  nil,
+			wantStatus:          framework.AsStatus(fmt.Errorf("running PreFilter plugin %q: %w", "error", errInjectedStatus)).WithFailedPlugin("error"),
+		},
+		{
+			name: "some PreFilter plugins returned skip",
+			plugins: []*TestPlugin{
+				{
+					name: "skip1",
+					inj:  injectedResult{PreFilterStatus: int(framework.Skip)},
+				},
+				{
+					name: "success1",
+				},
+				{
+					name: "skip2",
+					inj:  injectedResult{PreFilterStatus: int(framework.Skip)},
+				},
+				{
+					name: "success2",
+				},
+			},
+			wantPreFilterResult: nil,
+			wantSkippedPlugins:  sets.NewString("skip1", "skip2"),
+			wantStatus:          nil,
+		},
 	}
-	r := make(Registry)
-	r.Register(preFilterPluginName,
-		func(_ runtime.Object, fh framework.Handle) (framework.Plugin, error) {
-			return preFilter, nil
-		})
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := make(Registry)
+			enabled := make([]config.Plugin, len(tt.plugins))
+			for i, p := range tt.plugins {
+				p := p
+				enabled[i].Name = p.name
+				r.Register(p.name, func(_ runtime.Object, fh framework.Handle) (framework.Plugin, error) {
+					return p, nil
+				})
+			}
 
-	plugins := &config.Plugins{PreFilter: config.PluginSet{Enabled: []config.Plugin{{Name: preFilterPluginName}}}}
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
 
-	profile := config.KubeSchedulerProfile{Plugins: plugins}
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
+			t.Logf("name, %v", enabled)
 
-	f, err := newFrameworkWithQueueSortAndBind(r, profile, ctx.Done())
-	if err != nil {
-		t.Fatalf("Failed to create framework for testing: %v", err)
-	}
-	_, status := f.RunPreFilterPlugins(ctx, nil, nil)
-	wantStatus := framework.AsStatus(fmt.Errorf("running PreFilter plugin %q: %w", preFilter.Name(), errInjectedStatus)).WithFailedPlugin(preFilter.Name())
-	if !reflect.DeepEqual(status, wantStatus) {
-		t.Errorf("wrong status. got: %v, want:%v", status, wantStatus)
+			f, err := newFrameworkWithQueueSortAndBind(
+				r,
+				config.KubeSchedulerProfile{Plugins: &config.Plugins{PreFilter: config.PluginSet{Enabled: enabled}}},
+				ctx.Done(),
+			)
+			if err != nil {
+				t.Fatalf("Failed to create framework for testing: %v", err)
+			}
+
+			state := framework.NewCycleState()
+			result, status := f.RunPreFilterPlugins(ctx, state, nil)
+			if d := cmp.Diff(result, tt.wantPreFilterResult); d != "" {
+				t.Errorf("wrong status. got: %v, want: %v, diff: %s", result, tt.wantPreFilterResult, d)
+			}
+			if status == nil && tt.wantStatus != nil || status != nil && tt.wantStatus == nil {
+				t.Errorf("wrong status. got: %v, want: %v", status, tt.wantStatus)
+			} else if status == nil && tt.wantStatus == nil {
+				// do nothing
```

Comment:
doesn't `cmp.Diff` already handle all these cases?

Reply:
If status and/or tt.wantStatus is nil, `*tt.wantStatus` will be nil-pointer panic.

Reply:
you can use `cmp.Diff(status, tt.wantStatus)`

Reply:
I found ↑ is a lie. Actually, `cmp.Diff` can handle pointers.
The reason we cannot just use `cmp.Diff` is both errors are created from different error bases.

The status from TestPlugin contains the error created in `framework.NewStatus`.
https://github.com/kubernetes/kubernetes/blob/3efd107eb235d06feee95f80e461319d1dda7793/pkg/scheduler/framework/runtime/framework_test.go#L181
But, the status in each test case contains the error `errInjectedStatus`.
https://github.com/kubernetes/kubernetes/blob/3efd107eb235d06feee95f80e461319d1dda7793/pkg/scheduler/framework/runtime/framework_test.go#L382

---

I fixed here to use `cmp.Comparer` and deal with the ↑ problem.